dubbo-common/src/main/java/org/apache/dubbo/config/context/ConfigManager.java

```diff
@@ -197,6 +197,19 @@ public Optional<ProtocolConfig> getProtocol(String idOrName) {
         return getConfig(ProtocolConfig.class, idOrName);
     }
 
+    public ProtocolConfig getOrAddProtocol(String idOrName, boolean needRefresh) {
```

Comment:
When will needRefresh=false be used? I think it should always be refresh.

Reply:
Provides one more option, but it seems that no other place will call it at the moment, I will remove it.

Reply:
Already removed.